enclave/sgx/collateralinfo.c

```diff
@@ -461,13 +607,107 @@ void oe_free_sgx_quote_verification_collateral_args(
 {
     if (args)
     {
-        oe_free(args->tcb_info);
-        oe_free(args->tcb_info_issuer_chain);
-        oe_free(args->pck_crl);
-        oe_free(args->root_ca_crl);
-        oe_free(args->pck_crl_issuer_chain);
-        oe_free(args->qe_identity);
-        oe_free(args->qe_identity_issuer_chain);
-        oe_free(args->host_out_buffer);
+        if (args->tcb_info)
```

Comment:
Revert addition of if tests.  They increase code complexity, potentially reduce code coverage of tests, and provide no functional value here.